src/libs/actions/IOU.ts

```diff
@@ -7008,6 +7009,24 @@ function isLastApprover(approvalChain: string[]): boolean {
     return approvalChain[approvalChain.length - 1] === currentUserEmail;
 }
 
+function getNextApproverAccountID(report: OnyxEntry<OnyxTypes.Report>) {
+    const ownerAccountID = report?.ownerAccountID ?? -1;
+    const policy = PolicyUtils.getPolicy(report?.policyID);
+    const approvalChain = ReportUtils.getApprovalChain(policy, ownerAccountID, report?.total ?? 0);
+    const submitToAccountID = PolicyUtils.getSubmitToAccountID(policy, ownerAccountID);
+
+    if (approvalChain.length === 0) {
+        return submitToAccountID;
+    }
+
+    const nextApproverEmail = approvalChain.length === 1 ? approvalChain[0] : approvalChain[approvalChain.indexOf(currentUserEmail) + 1];
+    if (!nextApproverEmail) {
+        return submitToAccountID;
+    }
+
+    return PersonalDetailsUtils.getAccountIDsByLogins([nextApproverEmail])[0] ?? submitToAccountID;
```

Comment:
do we need to add a length check here?

Reply:
We don't need; there is already a fallback provided in case of empty array 👍